### PR TITLE
Fixes/rewrite to some operators, additional comments and additional parsers

### PR DESCRIPTION
**Pull request by deepsana**
## Summary of Changes
Fixes/rewrite to some operators — integral, summation, gradients
Parser can handle:
- Derivative has new enum component to identify various different notation. e.g. Leibniz, Newton, Euler, DNotation, Lagrange (Note that previous operator `PartialDerivative` is removed and partial derivative operator notation is now in `DerivativeNotation` enum. )
- Vector e.g. \vec{x}
### Related issues

Resolves ???

### CODE REVIEW

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 1]
```diff
@@ -6,28 +6,36 @@ use std::fmt;
...
 pub struct Derivative {
     pub order: u8,
     pub var_index: u8,
     pub bound_var: Ci,
 }
 
-/// Partial derivative operator
+/// D Derivative operator, e.g. DS/Dt . in line with Spivak notation: http://ceres-solver.org/spivak_notation.html
+#[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Clone, Hash, new, Deserialize, Serialize)]
+pub struct DDerivative {
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:20]: The `DDerivative` operator should be combined with the `Derivative` operator --- unless I am mistaken, their semantics are the same.

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 2]
```diff
@@ -6,28 +6,36 @@ use std::fmt;
...
-/// Summation operator with under and over components
-#[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Clone, Hash, new, Deserialize, Serialize, JsonSchema)]
-pub struct SumUnderOver {
-    pub op: Box<MathExpression>,
-    pub under: Box<MathExpression>,
-    pub over: Box<MathExpression>,
+/// Summation operator has the option of having lowlimit and uplimit components
+#[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Clone, Hash, new, Deserialize, Serialize)]
+pub struct Summation {
+    pub lowlimit: Option<Box<MathExpression>>,
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:37]: Can you change `lowlimit` to `lower_bound`

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 3]
```diff
@@ -6,28 +6,36 @@ use std::fmt;
...
+    pub uplimit: Option<Box<MathExpression>>,
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:38]: Can you change `uplimit` to `upper_bound`

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 4]
```diff
@@ -36,49 +44,60 @@ pub struct HatOp {
...
-/// Definite Integral with lowlimit, uplimit
-#[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Clone, Hash, new, Deserialize, Serialize, JsonSchema)]
-pub struct MsubsupInt {
-    pub lowlimit: Box<MathExpression>,
-    pub uplimit: Box<MathExpression>,
+/// Integral can be definite or indefinite with `integration_variable`
+/// as it has the option of having `lowlimit`, `uplimit`
+#[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Clone, Hash, new, Deserialize, Serialize)]
+pub struct Int {
+    pub lowlimit: Option<Box<MathExpression>>,
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:57]: Change `lowlimit` to `lower_limit`

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 5]
```diff
@@ -36,49 +44,60 @@ pub struct HatOp {
...
+    pub uplimit: Option<Box<MathExpression>>,
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:58]: Change `uplimit` to `upper_limit`

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 6]
```diff
@@ -100,30 +121,28 @@ pub enum Operator {
...
     Mean,
     /// Summation operator
-    Sum,
-    /// Summation operator with lowlimit and uplimit
-    SumUnderOver(SumUnderOver),
+    Summation(Summation),
+    /// Cross product operator
     Cross,
     /// Hat operator, e.g. \hat
     Hat,
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:131]: What does the `Hat` operator mean? and why is there a separate `HatOp` operator?

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 7]
```diff
@@ -100,30 +121,28 @@ pub enum Operator {
...
     HatOp(HatOp),
-    /// Summation operator with lowlimit and uplimit
-    MsupDownArrow(MsupDownArrow),
     /// ↓ as an operator
-    DownArrow,
-    /// Definite Integral
-    Int,
-    /// Indefinite Integral with lowlimit and uplimit
-    MsubsupInt(MsubsupInt),
+    DownArrow(DownArrow),
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:135]: What does the `DownArrow` operator mean?

File: skema/skema-rs/mathml/src/ast/operator.rs [hunk 8]
```diff
@@ -100,30 +121,28 @@ pub enum Operator {
...
+    /// Integrals
+    Int(Int),
+    /// Laplacian operator
     Laplacian,
     /// Closed surface integral operator --need to include explit dS integration variable when translating to latex
     SurfaceClosedInt,
     /// Closed surface integral operator -- doesn't need to include explicit dS integration variable when translating to latex
     /// E.g. \\oiint_S ∇ \cdot dS
     SurfaceClosedIntNoIntVar,
```
Comment (adarshp) [inline skema/skema-rs/mathml/src/ast/operator.rs:144]: `SurfaceClosedIntNoIntNoIntVar` should be combined with `SurfaceClosedInt`. Is there such a thing as an 'open' surface integral operator? If not, we may not need to specify 'closed' in the name of the operator.

**Review by adarshp — CHANGES_REQUESTED** [review]
I've left a few comments.

File: skema/skema-rs/skema/src/services/mathml.rs [hunk 9]
```diff
@@ -10,9 +10,9 @@ use mathml::parsers::math_expression_tree::{
 };
 use serde_json::from_str;
 use mathml::{
-    acset::{AMRmathml, PetriNet, RegNet, GeneralizedAMR},
-    parsers::first_order_ode::{first_order_ode, FirstOrderODE},
+    acset::{AMRmathml, PetriNet, RegNet},
     expression::get_code_exp_graphs,
+    parsers::first_order_ode::{first_order_ode, FirstOrderODE},
 };
```
Comment (Free-Quarks) [inline skema/skema-rs/skema/src/services/mathml.rs:16]: Just noting that this is likely why the tests aren't passing. The `GeneralizedAMR` import got deleted for some reason.

**Review by adarshp — APPROVED** [review]

**Comment by Free-Quarks** [comment]
This got messed up with outdated commits and we reverted to another branch that targeted a clean commit

**Pull request closed without merging**